src/main/java/Instruction.java

```diff
@@ -0,0 +1,46 @@
+// Handles the various input instructions for Yapper
+public class Instruction {
+    public enum InstructionType {
+        ADD, TODO, DEADLINE, EVENT, LIST, MARK, UNMARK, BYE
+    }
+    private InstructionType instructionType;
+    private String[] instructionStringArgs;
+    private String instructionStringArg;
+    private Integer instructionIntegerArg;
+
+    // Constructors
+    public Instruction(InstructionType type) {
+        this.instructionType = type; // LIST, BYE
+    }
+    public Instruction(InstructionType type, String arg) {
+        this.instructionType = type; // ADD, TODO
+        this.instructionStringArg = arg;
+    }
+    public Instruction(InstructionType type, String arg1, String arg2) {
+        this.instructionType = type; // DEADLINE
+        this.instructionStringArg = arg1;
+        this.instructionStringArgs = new String[] {arg2};
+    }
+    public Instruction(InstructionType type, String arg1, String arg2, String arg3) {
```

Comment:
Same thing, it would probably be more intuitive to to rename arg1, arg2 and arg 3 to something more descriptive.